### PR TITLE
Feat smartmeter sync time. Connects to #149

### PR DESCRIPTION
**Pull request by bjornamkreutz**
Added service contract: Synchronize Time, without testing. Connects to #149

### CODE REVIEW

**Comment by CLAAdmin** [comment]
[CLA assistant check](http://52.16.237.165:80/OSGP/Platform?pullRequest=158) <br/>You should sign our Contributor License Agreement in order to get your pull request merged.<br/>

**Comment by kevinsmeets** [comment]
Björn, please add headers for your code files.

/**
 * Copyright 2015 Smart Society Services B.V.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 */

File: osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/config/PersistenceConfig.java [hunk 1]
```diff
@@ -17,14 +17,22 @@
 import org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean;
 import org.springframework.transaction.annotation.EnableTransactionManagement;
 
+//<<<<<<< HEAD
```
Comment (thijlboonen) [inline osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/config/PersistenceConfig.java]: This merge conflict text has to be removed

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/mapping/InstallationMapper.java [hunk 2]
```diff
@@ -28,6 +28,6 @@ public void configure(final MapperFactory mapperFactory) {
         // domain value object -> dto value object
         mapperFactory
         .classMap(com.alliander.osgp.domain.core.valueobjects.smartmetering.SmartMeteringDevice.class,
-                com.alliander.osgp.dto.valueobjects.smartmetering.SmartMeteringDevice.class).byDefault().register();
```
Comment (thijlboonen) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/mapping/InstallationMapper.java]: This has to be reverted.

This maps a domain value object -> domain value object. It also breaks the add meter functionality.

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/services/InstallationService.java [hunk 3]
```diff
@@ -110,8 +110,8 @@ public void addMeter(
             throw new FunctionalException(FunctionalExceptionType.EXISTING_DEVICE, ComponentType.DOMAIN_SMART_METERING);
         }
 
-        final com.alliander.osgp.dto.valueobjects.smartmetering.SmartMeteringDevice smartMeteringDevicDto = this.installationMapper
```
Comment (thijlboonen) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/services/InstallationService.java]: This has to be reverted. It breaks the addMeter functionality.

File: osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/services/AdhocService.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+    	
+        LOGGER.debug("enqueueSynchronizeTimeReadsRequest called with organisation {} and device {}",
+                organisationIdentification, deviceIdentification);
+
+        final String correlationUid = this.correlationIdProviderService.getCorrelationId(organisationIdentification,
+                deviceIdentification);
+
+        final SmartMeteringRequestMessage message = new SmartMeteringRequestMessage(
+                SmartMeteringRequestMessageType.REQUEST_PERIODIC_METER_DATA, correlationUid, organisationIdentification,
```
Comment (thijlboonen) [inline osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/services/AdhocService.java]: Shouldn't this be REQUEST_SYNCHRONIZE_TIME?

File: osgp-adapter-ws-smartmetering/src/main/webapp/WEB-INF/wsdl/smartmetering/SmartMeteringAdhoc.wsdl [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        </xsd:schema>
+    </wsdl:types>    
+
+    <wsdl:message name="ActionHeader">
+        <wsdl:part element="common:OrganisationIdentification" name="OrganisationIdentification" />
+        <wsdl:part element="common:UserName" name="UserName" />
+		<wsdl:part element="common:ApplicationName" name="ApplicationName" />
+    </wsdl:message>
+    
+    <!-- REQUEST_PERIODIC_METER_DATA -->
```
Comment (thijlboonen) [inline osgp-adapter-ws-smartmetering/src/main/webapp/WEB-INF/wsdl/smartmetering/SmartMeteringAdhoc.wsdl]: Change to REQUEST_SYNCHRONIZE_TIME

**Comment by thijlboonen** [comment]
Also remove all @author OSGP comments and format the code with save actions.

File: osgp-core/.gitignore [hunk 6]
```diff
@@ -0,0 +1 @@
+/target/
```
Comment (kevinsmeets) [inline osgp-core/.gitignore]: Please remove this git ignore file.

File: osgp-adapter-ws-smartmetering/src/main/webapp/WEB-INF/wsdl/smartmetering/SmartMeteringAdhoc.wsdl [hunk 7]
```diff
@@ -0,0 +1,104 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (kevinsmeets) [inline osgp-adapter-ws-smartmetering/src/main/webapp/WEB-INF/wsdl/smartmetering/SmartMeteringAdhoc.wsdl]: Please add header to this file.

File: osgp-adapter-ws-smartmetering/src/main/resources/db/migration/V2015.006__Added_synchronize_time_table.sql [hunk 8]
```diff
@@ -0,0 +1,28 @@
+CREATE TABLE synchronize_time_reads (
```
Comment (kevinsmeets) [inline osgp-adapter-ws-smartmetering/src/main/resources/db/migration/V2015.006__Added_synchronize_time_table.sql]: Please remove this sql script.

File: osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/domain/repositories/SynchronizeTimeDataRepository.java [hunk 9]
```diff
@@ -0,0 +1,19 @@
+/**
```
Comment (kevinsmeets) [inline osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/domain/repositories/SynchronizeTimeDataRepository.java]: Please remove this repository.

File: osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/domain/entities/SynchronizeTimeReads.java [hunk 10]
```diff
@@ -0,0 +1,43 @@
+package com.alliander.osgp.adapter.ws.smartmetering.domain.entities;
```
Comment (kevinsmeets) [inline osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/domain/entities/SynchronizeTimeReads.java]: Please remove this entity.

File: osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/services/AdhocService.java [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+                deviceIdentification, requestData);
+
+        this.smartMeteringRequestMessageSender.send(message);
+
+        return correlationUid;
+    }
+
+    /**
+     * @param organisationIdentification
+     * @param device
```
Comment (kevinsmeets) [inline osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/services/AdhocService.java]: Please fix or remove this comment.

**Comment by kevinsmeets** [comment]
Please move all headers in code files to the top of the file.

**Pull request merged**